R/consolidate_metadata.R

```diff
@@ -0,0 +1,125 @@
+#' Consolidate Multiple Meta Datasets Into a Single One
+#'
+#' The purpose of the function is to consolidate multiple meta datasets into a
+#' single one. For example, from global and project specific parameter mappings
+#' a single lookup table can be created.
+#'
+#' @param datasets List of datasets to consolidate
+#'
+#'   *Permitted Values*: A named list of datasets
+#'
+#' @param key_vars Key variables
+#'
+#'   The specified variables must be a unique of all input datasets.
+#'
+#'   *Permitted Values*: A list of variables created by `vars()`
+#'
+#' @param source_var Source variable
+#'
+#'   The specified variable is added to the output dataset. It is set the name
+#'   of the dataset the observation is originating from.
+#'
+#'   *Permitted Values*: A symbol
+#'
+#' @param check_vars Check variables?
+#'
+#'   If `"message"`, `"warning"`, or `"error"` is specified, a message is issued
+#'   if the variable names differ across the input datasets (`datasets`).
+#'
+#'   *Permitted Values*: `"none"`, `"message"`, warning"`, `"error"`
+#'
+#' @param check_keys Check keys?
+#'
+#'   If `"warning"` or `"error"` is specified, a message is issued if the key
+#'   variables (`key_vars`) are not a unique key in all of the input datasets
+#'   (`datasets`).
+#'
+#'   *Permitted Values*: `"none"`, `"warning"`, `"error"`
+#'
+#' @author Stefan Bundfuss
+#'
+#' @details All observations of the input datasets are put together into a
+#'   single dataset. If a by group (defined by `key_vars`) exists in more than
+#'   one of the input datasets, the observation from the last dataset is
+#'   selected.
+#'
+#' @return A dataset which contains one row for each by group occuring in any of
+#'   the input datasets.
+#'
+#' @keywords create_aux
+#' @family create_aux
```

Comment:
Can be confusing if the word "metadata" is in the function name but not under the Metadata section of our reference page

Reply:
Yes, but the "Metadata" section lists datasets and not functions. The description says "Auxiliary datasets providing definitions as input for derivations, e.g. grading criteria or dose frequencies". Thus `consolidate_metadata()` does not really fit into this section. I think it fits better into "Creating Auxiliary Datasets" although I must admit that it is not ideal.